WordPress/gutenberg
# [ experimental/nav-menus ] Delete menus in nav menus screen

## Description
Adds delete button to #21281

Given deleting entities is a complex endeavor, that will eventually allow to undo deletes before saving, this approach advances the navigation screen and allows for user testing.

It is not the best way to implement this, but it allows the new screen to have this behavior now and not wait on some technicalities to be solved.

## How has this been tested?
Tested locally by:

 - Enable the menu edit experiment
 - Make sure to use a theme that supports menus
 - Create some menus
 - Go to Gutenberg -> Navigation (beta)
 - Start deleting menus


## Screenshots 

![delete-menus](https://user-images.githubusercontent.com/107534/78794484-7c090580-79bc-11ea-87d6-35cced6cab9d.gif)

## Types of changes
Non breaking change that adds a delete call to the API inside the `MenuEditor` component and creates a state from initial menus inside `MenusEdtior` component.
